innovation.game.php

```diff
@@ -12786,6 +12904,52 @@ function stInterInteractionStep() {
                     }
                     break;
 
+                // id 196, Artifacts age 9: Luna 3
+                case "196N1A":
+                    // Draw and score a card of value equal to the number of cards returned.
+                    self::executeDraw($player_id, $n, 'score');
+                    break;
+
+                // id 200, Artifacts age 9: Syncom 3
+                case "200N1A":
+                    // "Draw and reveal five 9."
+                    $color_detect = array(0,0,0,0,0);
+                    $card_array = array();
+                    $repeat_colors = false;
+                    for($card_idx = 0; $card_idx < 5; $card_idx++) {
+                        $card = self::executeDraw($player_id, 9, 'revealed');
+                        if ($color_detect[$card['color']] == 1) { 
+                            $repeat_colors = true; // if the flag is 1, the color already has been revealed
+                        }
+                        $color_detect[$card['color']] = 1; // flag the color
+                        $card_array[] = $card;
+                    }
+                    
+                    // If you revealed all five colors, you win.
+                    if ($repeat_colors == false) {
```

Comment:
I was able to simplify this a lot by using array_count_values to help count the unique values.